v0.2.0 [rc]

SebanSunny opened the pull request:
- updated KBSearch with the cosine similarity implementation and variable thresholds; updated the module to check for results from QD module.
- updated Comprehension to check for 'on_topic' flag from KBSearch instead of 'kb_article'.

dwhitena commented on kb-search/requirements.txt:
```diff
@@ -1,3 +1,3 @@
-nltk==3.4.1
```
Could you go ahead and make sure all NLTK version throughout Katecheo are upgraded as prescribed here: https://github.com/cvdigitalai/katecheo/pull/17.

dwhitena commented on target-classifier/TargetClassifier.py:
```diff
@@ -71,7 +71,7 @@ def predict(self, X, feature_names, meta):
             matchedEntities = []
 
             # Get the text string that is to be classified.
-            messageText = str(X[0])
```
Aren't we removing target classifier? Shouldn't the logic now be:

(question detector) -> (KB search) -> (comprehension)

I think this will require a change to the deployement JSON, docs, and removal of some code right?

dwhitena commented:
I'm getting the following error when trying to launch the KB search model server:

```
2019-12-11 11:41:41,862 - seldon_core.microservice:main:189 - INFO:  Importing KBSearch
Traceback (most recent call last):
  File "/usr/local/bin/seldon-core-microservice", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/seldon_core/microservice.py", line 202, in main
    user_object = user_class(**parameters)
  File "/microservice/KBSearch.py", line 62, in __init__
    if os.environ['ARTICLE_ID'] and os.environ[
  File "/usr/local/lib/python3.6/os.py", line 669, in __getitem__
    raise KeyError(key) from None
KeyError: 'ARTICLE_ID'```

Do we need to update the knowledge bases and then test?

dwhitena commented:
Ok @SebanSunny, I used the article links in the unit test and that fixed the above error. However, now I'm getting:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/nltk/corpus/util.py", line 86, in __load
    root = nltk.data.find('{}/{}'.format(self.subdir, zip_name))
  File "/usr/local/lib/python3.6/site-packages/nltk/data.py", line 701, in find
    raise LookupError(resource_not_found)
LookupError:
**********************************************************************
  Resource stopwords not found.
  Please use the NLTK Downloader to obtain the resource:

  >>> import nltk
  >>> nltk.download('stopwords')

  For more information see: https://www.nltk.org/data.html

  Attempted to load corpora/stopwords.zip/stopwords/

  Searched in:
    - '/root/nltk_data'
    - '/usr/local/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/local/lib/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/local/lib/nltk_data'
**********************************************************************
```

When I run the following locally (to build the image locally):

```
$ s2i build . seldonio/seldon-core-s2i-python3:0.7 kb-search-test
$ docker run --env KATECHEO_KB='faith=https://storage.googleapis.com/pachyderm-neuralbot/knowledge_bases/kb_gq_6k_general.json,health=https://storage.googleapis.com/pachyderm-neuralbot/knowledge_bases/kb_wedmd_health.json' --env ARTICLE_ID="article_url" --env ARTICLE_TITLE_KEY="title" --env ARTICLE_BODY_KEY="body" --env COSINE_SIMILARITY_THRESHOLD="0.19" --name "kb-search" --rm -p 5001:5000 kb-search-test
```

SebanSunny commented:
@dwhitena NLTK had some data dependencies for the following packages:
- stopwords
- punkt

I've added this to the KBSearch file.
I've also updated the steps for KBSearch in the readme

dwhitena commented:
@SebanSunny All looks great to me now!

SebanSunny commented:
@dwhitena Something that I noticed today was that we need to all files in the deployment folder with the new environment variables required by KB Search...
